spec.emu

```diff
@@ -268,93 +271,204 @@ contributors: Ujjwal Sharma, Younies Mahmoud
+        1. Return _result_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-formatseconds" type="abstract operation">
+      <h1>
+        FormatNumericSeconds (
+          _durationFormat_: a DurationFormat Object,
+          _secondsValue_: an integer,
+          _minutesDisplayed_ : a Boolean,
+          _signDisplayed_: a Boolean,
+          ) : a List of Records
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It creates a List of Records containing the formatted parts of the seconds unit of _duration_ and all subsecond units when the requested style is *"numeric"* or *"2-digit"* according to the effective locale and the formatting options of _durationFormat_.</dd>
+      </dl>
+
+      <emu-alg>
+        1. Let _result_ be a new empty List.
+        1. If _minutesDisplayed_ is *true*, then
+          1. Let _digitalFormat_ be _durationFormat_.[[DigitalFormat]].
+          1. Let _separator_ be _digitalFormat_.[[MinutesSecondsSeparator]].
+          1. Append the Record { [[Type]]: *"literal"*, [[Value]]: _separator_, [[Unit]]: ~empty~ } to _result_.
+        1. Let _nfOpts_ be OrdinaryObjectCreate(*null*).
+        1. Let _secondsStyle_ be _durationFormat_.[[SecondsStyle]].
+        1. Assert: _secondsStyle_ is *"numeric"* or _secondsStyle_ is *"2-digit"*.
+        1. If _secondsStyle_ is *"2-digit"*, then
+          1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"minimumIntegerDigits"*, *2*<sub>𝔽</sub>).
+          1. If _signDisplayed_ is *false*, then
+            1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"signDisplay"*, *"never"*).
+        1. Let _nf_ be ! Construct(%NumberFormat%, &laquo; _durationFormat_.[[Locale]], _nfOpts_ &raquo;).
+        1. Let _fractionalDigits_ be AddFractionalDigits(_durationFormat_, _duration_).
+        1. If _fractionalDigits_ is not 0, then
+          1. If _durationFormat_.[[FractionalDigits]] is *undefined*, then
+            1. Let _maximumFractionDigits_ be *9*<sub>𝔽</sub>.
+            1. Let _minimumFractionDigits_ be *+0*<sub>𝔽</sub>.
           1. Else,
-            1. Set _needSeparator_ to *false*.
+            1. Let _maximumFractionDigits_ be _durationFormat_.[[FractionalDigits]].
+            1. Let _minimumFractionDigits_ be _durationFormat_.[[FractionalDigits]].
+          1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"maximumFractionDigits"*, _maximumFractionDigits_).
+          1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"minimumFractionDigits"*, _minimumFractionDigits_).
+          1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"roundingMode"*, *"trunc"*).
+        1. Let _secondsParts_ be ! PartitionNumberPattern(_nf_, _secondsValue_).
+        1. For each Record { [[Type]], [[Value]] } _part_ of _secondsParts_, do
+          1. Append the Record { [[Type]]: _part_.[[Type]], [[Value]]: _part_.[[Value]], [[Unit]]: *"second"* } to _result_.
+        1. Return _result_.
+        </emu-alg>
+    </emu-clause>
```

Comment:
`FormatNumericHours` and `FormatNumericMinutes` are combinable, but `FormatNumericSeconds` is pretty different from both due to the extra properties it has to set on `_nfOpts_`. It could save some repeated code to combine the first two into `FormatNumericHoursMinutes`, though -- I'll give it a try.